scala-generator/src/main/scala/models/generator/ScalaClientMethodConfig.scala

```diff
@@ -218,16 +228,18 @@ private lazy val defaultAsyncHttpClient = PooledHttp1Client()
   }
 
   case class Http4s018(namespace: String, baseUrl: Option[String]) extends Http4s {
-    override val asyncType = "cats.effect.IO"
+    override val asyncType = "F"
+    override val asyncTypeParam = Some(s"[$asyncType[_]: Effect]")
+    override val asyncTypeMultipleParam = Some(s"$asyncType[_]: Effect")
     override val leftType = "Left"
     override val rightType = "Right"
     override val monadTransformerInvoke = "value"
     override val responseClass = s"org.http4s.Response[$asyncType]"
-    override val extraClientCtorArgs = Some(s",\n  asyncHttpClient: org.http4s.client.Client[$asyncType] = Client.defaultAsyncHttpClient")
+    override val extraClientCtorArgs = Some(s",\n  asyncHttpClient: org.http4s.client.Client[$asyncType] = Client.defaultAsyncHttpClient[$asyncType]")
     override val extraClientObjectMethods = Some(s"""
-implicit def circeJsonDecoder[A](implicit decoder: io.circe.Decoder[A]) = org.http4s.circe.jsonOf[$asyncType, A]
+implicit def circeJsonDecoder[${asyncTypeMultipleParam.map(_+", ").getOrElse("")}A](implicit decoder: io.circe.Decoder[A]) = org.http4s.circe.jsonOf[$asyncType, A]
 
-private lazy val defaultAsyncHttpClient = PooledHttp1Client[$asyncType]()
+private def defaultAsyncHttpClient${asyncTypeParam.getOrElse("")} = PooledHttp1Client[$asyncType]()
```

Comment:
We don't have a clean way of shutting this down (and could end up with lots of independent pools running around). I would suggest killing all of this, and make the user pass in an `org.http4s.client.Client` themselves.